pkg/keyspace/keyspace.go

```
@@ -1035,33 +1050,77 @@ func (manager *Manager) UpdateKeyspaceState(name string, newState keyspacepb.Key
+// UpdateKeyspaceInfoMetrics updates the keyspace ID-to-name mapping metric when keyspace-level metrics are enabled.
+func (manager *Manager) UpdateKeyspaceInfoMetrics(meta *keyspacepb.KeyspaceMeta) {
+	keyspaceInfoMetricsMu.Lock()
+	defer keyspaceInfoMetricsMu.Unlock()
+	if meta == nil || !manager.config.IsKeyspaceLevelMetricsEnabled() {
+		return
+	}
+	setKeyspaceInfoMetricsLocked(meta.GetId(), meta.GetName())
+}
+
+// DeleteKeyspaceInfoMetrics removes the cached keyspace info series for id.
+func (manager *Manager) DeleteKeyspaceInfoMetrics(id uint32) {
```

golangci-lint (2.12.2)

[warning] unused-receiver: method receiver 'manager' is not referenced in method's body, consider removing or renaming it as _

(revive)